x-pack/legacy/plugins/canvas/public/state/reducers/embeddable.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { fromExpression, toExpression } from '@kbn/interpreter/common';
+import { handleActions } from 'redux-actions';
+import { State } from '../../../types';
+
+import {
+  UpdateEmbeddableExpressionActionType,
+  UpdateEmbeddableExpressionPayload,
+} from '../actions/embeddable';
+
+// @ts-ignore untyped local
+import { assignNodeProperties } from './elements';
+
+export const embeddableReducer = handleActions<
+  State['persistent']['workpad'],
+  UpdateEmbeddableExpressionPayload
+>(
+  {
+    [UpdateEmbeddableExpressionActionType]: (workpadState, { payload }) => {
+      if (!payload) {
+        return workpadState;
+      }
+
+      const { elementId, embeddableExpression } = payload;
+
+      // Find the element
+      const pageWithElement = workpadState.pages.find(page => {
```

Comment:
nit: we're doin a lot of this kind of finding a page or element given an element id across this code. maybe something to refactor?